packages/@aws-cdk/aws-events-targets/test/ecs/integ.event-fargate-task.expected.json

```diff
@@ -609,7 +609,7 @@
                     "Arn"
                   ]
                 },
-                "Id": "TaskDef-on-EcsCluster",
+                "Id": "awsecsintegfargateTaskDef8878AF94-on-awsecsintegfargateEcsCluster46E816CB",
```

Comment:
The custom resource will not work anymore now because the `Id` here is not the same as the real `Id` of the target: `awsecsintegfargateTaskDef8878AF94-on-awsecsintegfargateEcsCluster46E816CB` vs `egfargateTaskDef8878AF94-on-awsecsintegfargateEcsCluster46E816CB`

Reply:
😞I think you are right, need to be changed I guess

Reply:
The test here should also check that both `Custom::AWS` and `AWS::Events::Rule` have the same id https://github.com/awslabs/aws-cdk/blob/696f53f8ee426a7a7958d0807c7ad89b72f587ae/packages/%40aws-cdk/aws-events-targets/test/ecs/event-rule-target.test.ts#L59

Reply:
> The custom resource will not work anymore now because the `Id` here is not the same as the real `Id` of the target: `awsecsintegfargateTaskDef8878AF94-on-awsecsintegfargateEcsCluster46E816CB` vs `egfargateTaskDef8878AF94-on-awsecsintegfargateEcsCluster46E816CB`

I don't know if it makes sense but... for this, I made some reasoning about and I thought that maybe we could just use the `this.taskDefinition.node.uniqueId`: this is because if you look in `Construct` ([source](https://github.com/awslabs/aws-cdk/blob/e4fb811c5b12de4d42fea71510e5bb3ee6b0a73e/packages/%40aws-cdk/cdk/lib/construct.ts#L108)), the `uniqueId` is guaranteed to be *unique* across the same app - and this should be enough to satisfy the constraint of being *scope-unique*. If this makes sense, then the same reasoning should be valid for the `cluster.node` and this could be used as well, but the taskDefinition fits better imho. Am I wrong? 

> The test here should also check that both Custom::AWS and AWS::Events::Rule have the same id

For this, I'm not sure if it is already ok the test (it works, but I'm not convinced): my doubt are about the transformation. I mean, for the custom resource do we have somehow to assume how the id is *sanitized*? If yes, then maybe is not ok yet and it has to be changed because it doesn't cover the scenario you described...but tbh, I'm not sure 🤔

Reply:
Using `this.taskDefinition.node.uniqueId` seems a good solution. 

Even if the `uniqueId` is guaranteed to be unique across the app note that in the rare event that someone would want to use the same task definition in a different cluster but in the same rule it won't work... don't think it's problematic.